### ISSUE TITLE
`unused-parameter` rule should not trigger on `override` functions

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#5743]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

`override` function have to perfectly match the superclass, so you cannot change or remove the parameter then.

Example:

```swift
final class Foo: NSView {
	override func acceptsFirstMouse(for event: NSEvent?) -> Bool { true }
}
```

Same applies to protocol conformances, but I don't think it's possible to catch such cases unless the rule becomes an analyzer rule.

### Environment

* SwiftLint version: 0.56.0
* Xcode version: Xcode 15.4, Build version 15F31d
* Installation method used: Installer
* Configuration file:

```yml
only_rules:
  - unused_parameter
```

**Comment by SimplyDanny** [comment]
What the rule suggests in this case is
```swift
final class Foo: NSView {
	override func acceptsFirstMouse(for _: NSEvent?) -> Bool { true }
}
```
which is even valid for `override` functions.

**Comment by sindresorhus** [comment]
That's not optimal, but I guess this can be closed in favor of:

```swift
final class Foo: NSView {
	override func acceptsFirstMouse(for _event: NSEvent?) -> Bool { true }
}
```

https://github.com/realm/SwiftLint/issues/5741#issuecomment-2280811785
